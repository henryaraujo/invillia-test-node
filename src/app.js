import 'dotenv/config';
import koa from 'koa';
import json from 'koa-json';
import body from 'koa-body';
import jwt from 'koa-jwt';

import router from '../src/routes';

const app = new koa();

const secret = process.env.JWT_SECRET || 'aW52aWxsaWEtdGVzdC1ub2Rl';

app.use(body({ 
    multipart: true,
    urlencoded: true
}))
.use(json({ pretty: true, param: 'pretty' }))
.use(jwt({ secret: secret }).unless({ path: [/^\/v1\/api\/auth/] }))
.use(router.routes())
.use(router.allowedMethods());

export default app