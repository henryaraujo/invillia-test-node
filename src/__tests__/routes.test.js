import request from 'supertest';
import app from '../app';

const URL = '/v1/api/';
var token = '';

describe('POST /auth/signin', () => {

    test('expect token user', async () => {
        const user = {
            email: 'henryaraujo@email.com',
            password: '123456'
        }
        const res = await request(app.callback()).post(`${URL}auth/signin`).send(user);
        expect(res.status).toBe(200);
        expect(res.body).toHaveProperty("token");
        token = res.body.token
    });
})



describe('GET routes api', () => {

    test('expect list player', async () => {
        const res = await request(app.callback())
                        .get(`${URL}players`)
                        .set('Authorization', `Bearer ${token}`)
        expect(res.status).toBe(200);
        expect(res.type).toEqual("application/json");
        expect(res.body).toEqual(
            expect.arrayContaining([]),
        );
    });


    test('expect list tournaments', async () => {
        const res = await request(app.callback())
                        .get(`${URL}tournaments`)
                        .set('Authorization', `Bearer ${token}`)
        expect(res.status).toBe(200);
        expect(res.body).toEqual(
            expect.arrayContaining([]),
        );
    });


    test('expect list phases', async () => {
        const res = await request(app.callback())
                        .get(`${URL}phases`)
                        .set('Authorization', `Bearer ${token}`)
        expect(res.status).toBe(200);
        expect(res.body).toEqual(
            expect.arrayContaining([]),
        );
    });


    test('expect list results', async () => {
        const res = await request(app.callback())
                        .get(`${URL}phases`)
                        .set('Authorization', `Bearer ${token}`)
        expect(res.status).toBe(200);
        expect(res.body).toEqual(
            expect.arrayContaining([]),
        );
    });
})


describe('POST routes api', () => {

    test('create new player', async () => {
        const player = {
            name: "Stewie", 
            nicknam: "stewk2", 
            level: 20,
            is_active: true
        }
        const res = await request(app.callback())
                        .post(`${URL}players`)
                        .set('Authorization', `Bearer ${token}`)
                        .send(player);
        expect(res.status).toBe(200);
        expect(res.body).toHaveProperty("_id");
    });


    test('create new tournament', async () => {
        const player ={
            name: "CS GO for Newbies", 
            description: "Rank Prata mode=on", 
            year: "2019",
            game: "CS GO",
            is_active: true,
            phases : [
                "Dq7OJv7fkhJWpYoi",
                "YCvs2w4NqfWHS8ZY"
            ]
        }
        const res = await request(app.callback())
                        .post(`${URL}tournaments`)
                        .set('Authorization', `Bearer ${token}`)
                        .send(player);
        expect(res.status).toBe(200);
        expect(res.body).toHaveProperty("_id");
    });

    test('create new phase', async () => {
        const phase = {
            name: "Oitavas",
            description: "Melhor de 3",
            max_players: 10,
            is_active: true,
            players: [
                "Zva5rKoAg2qvzrIt",
                "l5ZndOcwx4ZZgEOc",
                "mn6VWYysfQdOvLlN"
             ]
        }
        const res = await request(app.callback())
                        .post(`${URL}phases`)
                        .set('Authorization', `Bearer ${token}`)
                        .send(phase);
        expect(res.status).toBe(200);
        expect(res.body).toHaveProperty("_id");
    });

    test('create new results', async () => {
        const result ={
            player_id: "sN0sMEhn4H3NabfT", 
            phase_id: "Dq7OJv7fkhJWpYoi", 
            points: 80,
            is_active: true
        }
        const res = await request(app.callback())
                        .post(`${URL}phases`)
                        .set('Authorization', `Bearer ${token}`)
                        .send(result);
        expect(res.status).toBe(200);
        expect(res.body).toHaveProperty("_id");
    });
})

