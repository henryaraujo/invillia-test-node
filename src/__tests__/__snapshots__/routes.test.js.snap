// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`POST /auth/signin expect token user 1`] = `
"{
  \\"token\\": \\"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJuYW1lIjoiSGVucmlxdWUgQXJhdWpvIiwiZW1haWwiOiJoZW5yeWFyYXVqb0BlbWFpbC5jb20iLCJfaWQiOiJMNUc1Y050Yld1UGtSRHlmIiwiaWF0IjoxNTYwNTM5NTc5fQ.OP8uRaX5-4T9deF6O3EqCA9H8xDf_cCBXmin4YZCqLc\\"
}"
`;
